config/v1/types_infrastructure.go

```diff
@@ -705,7 +705,45 @@ type AlibabaCloudResourceTag struct {
 
 // NutanixPlatformSpec holds the desired state of the Nutanix infrastructure provider.
 // This only includes fields that can be modified in the cluster.
-type NutanixPlatformSpec struct{}
+type NutanixPlatformSpec struct {
+	// prismCentral holds the endpoint address and port to access the Nutanix Prism Central
+	// +kubebuilder:validation:Required
+	PrismCentral NutanixPrismEndpoint `json:"prismCentral"`
+
+	// prismElements holds one or more endpoint address and port data to access the Nutanix
+	// Prism Elements (clusters) of the Nutanix Prism Central. Currently we only support one
+	// Prism Element (cluster) for an openshift cluster, where all the Nutanix resources (VMs, subnet, etc.)
+	// used in the Openshift cluster locate. In the future, we may support the Nutanix resources (VMs, etc.)
+	// used in the Openshift cluster can come from multiple Prism Elements (clusters) of the Prism Cental.
+	// +kubebuilder:validation:Required
+	PrismElements []NutanixPrismElementEndpoint `json:"prismElements"`
+}
+
+// NutanixPrismEndpoint holds the endpoint address and port to access the Nutanix Prism Central or Element (cluster)
+type NutanixPrismEndpoint struct {
+	// address is the endpoint address (DNS name or IP address) of the Nutanix Prism Central or Element (cluster)
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength=256
+	Address string `json:"address"`
+
+	// port is the port number to access the Nutanix Prism Central or Element (cluster)
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=65535
+	Port int32 `json:"port"`
+}
+
+// NutanixPrismElementEndpoint holds the name and endpoint data for a Prism Element (cluster)
+type NutanixPrismElementEndpoint struct {
+	// name is the name of the Prism Element (cluster)
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength=256
+	Name string `json:"name"`
```

Comment:
I'm guessing that this will tie to some object field, so in the machine providerSpec will it have an element name to be deployed to? Likewise, will the CSI storage have a way to configure which element name to be deployed to?

Reply:
It would be good if you could expand this comment to explain that other resources will have a cluster field which will reference a name within this list. The context you've just replied here is very helpful for the user to understand how this value relates to their other resources

Reply:
All the configuration details (such as this) will be document in the installation document.

Reply:
This is also user facing documentation and a note should be added here as well IMO. Only needs to be something along the lines of "This value will correspond with the `cluster` field configured on other resources (eg Machines, PVCs, etc)"